_posts/2025-04-09-#31664.md

```diff
@@ -10,17 +10,120 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+### Motivation and Background
 
+The current Fee Estimator in Bitcoin Core ([`CBlockPolicyEstimator`](https://github.com/bitcoin/bitcoin/blob/4c1906a500cacab385b09e780b54271b0addaf4b/src/policy/fees.h#L148)) used by the wallet and exposed to clients via RPC, has two key limitations that affect users:
 
-## Questions
+1. **Mempool unaware**: Does not consider the current mempool unconfirmed transactions when estimating fees. As a result, it remains oblivious to sudden changes, such as when high fee rate transactions are cleared and the mempool becomes sparse, leading to overpayment. Likewise, during sudden congestion, it fails to adjust, resulting in underpayment and missed confirmation targets.
+
+2. **Package unaware**: Only considers individual transactions, ignoring parent-child relationships. As a result, it fails to account for [CPFP'd](https://bitcoinops.org/en/topics/cpfp/) transactions, which can lead to lower fee rate estimates, resulting in underpayment and missed confirmation targets.
+
+
+**These limitations cause practical problems for:**  
+- **Bitcoin Core wallet users** ([many now rely on third-party providers](https://hackmd.io/@kEyqkad6QderjWKtcBF5Hg/cChallengies-with-estimating-transaction-fees#Analysis-to-Confirm-that-CBlockPolicyEstimator-is-unreliable-in-some-situations)) due to the inaccuracies mentioned above.
+- **Lightning implementations** (e.g., C-lightning), which depend on Bitcoin Core's fee estimator, may construct HTLC transactions that fail to meet deadlines or pay more than necessary.
+- **Bitcoin clients** ([e.g., BTCPay Server](https://github.com/btcpayserver/btcpayserver/pull/5490)), have switched to external fee estimators due to Bitcoin Core's mempool unawareness. 
+- **Others** ([e.g., Decred DEX](https://github.com/decred/dcrdex/pull/3046)), continue to see high fee rate estimates more than necessary even after the recent v29 [switching of estimateSmartFee mode to "economical"](https://github.com/bitcoin/bitcoin/pull/30275).
+
+Bitcoin Core needs to provide reliable fee rate estimates to uphold the trustlessness and self-sovereignty of node operators and the services that rely on it. Relying on external estimators undermines these principles. Bitcoin Core’s fee estimator should be as reliable and cost effective as external alternatives.
```

Comment:
what do you mean by cost effective?

Reply:
From Britacanna https://www.britannica.com/dictionary/cost%E2%80%93effective, I just meant users should not have to pay more than necessary.